### PR TITLE
patron: improve patron brief view

### PR DESCRIPTION
**Pull request by zannkukai**
* Makes the `circulation` link as any other action button.
* Use badges to display patron roles.

## How to test 
Before:
image

After:
image

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+-->
+<div *ngIf="record"
+     [attr.id]="record.metadata.pid | idAttribute : {'prefix': 'patron', 'suffix': 'brief-view'}"
+     class="patron-brief brief-view">
+  <h5 class="card-title">
+    <a [routerLink]="[detailUrl.link]">
+      {{ record.metadata.last_name }}<ng-container *ngIf="record.metadata.first_name">, {{ record.metadata.first_name }}</ng-container>
+    </a>
+    <small class="badge ml-2" [class]="getBadgeColor(role)" *ngFor="let role of record.metadata.roles">
+      {{ role | translate }}
```
Comment (jma) [inline projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.html:26]: translate can be specified in the `small` tag.

File: projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <div *ngIf="record.metadata.patron" class="float-right mb-2 mr-n5">
+      <button class="btn btn-sm btn-outline-primary ml-2" [routerLink]="['/circulation', 'patron', record.metadata.patron.barcode[0]]">
+        <i class="fa fa-exchange mr-2"></i>
+        <span translate>Circulation</span>
+      </button>
+    </div>
+  </h5>
+  <div class="card-text">
+    <p>{{ record.metadata.birth_date | dateTranslate:'mediumDate' }} &mdash; {{ record.metadata.city }}</p>
+    <!--<ng-container *ngIf="record.metadata.roles.length > 0">
```
Comment (jma) [inline projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.html:37]: can be removed

File: projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  @Input() type: string;
+  /** the url to access detail view */
+  @Input() detailUrl: { link: string, external: boolean };
+
+  /** mapping to define role badge color */
+  private _rolesBadgeMappings = {
+    system_librarian: 'badge-danger',
+    librarian: 'badge-warning'
+  };
+
```
Comment (jma) [inline projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.ts:40]: too much empty lines

File: projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+  getBadgeColor(role: string): string {
```
Comment (jma) [inline projects/admin/src/app/record/brief-view/patrons-brief-view/patrons-brief-view.component.ts:43]: documentation is missing

File: projects/admin/src/app/scss/variables.scss [hunk 5]
```diff
@@ -33,3 +33,5 @@ $form-check-input-margin-y:           .2rem !default;
 $form_error: #a94442;
 $form_required: red;
 $form_valid: green;
+
+$light: #EEEEEE;
```
Comment (jma) [inline projects/admin/src/app/scss/variables.scss:37]: Why not using `$gray-100` or `$gray-xxx`?

**Review by jma — APPROVED** [review]

**Pull request merged**